src/app/shared/services/children/children.service.ts

```diff
@@ -1,28 +1,30 @@
+import { Observable } from 'rxjs';
+
 import { HttpClient, HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Store } from '@ngxs/store';
-import { Observable } from 'rxjs';
+
 import { Child, ChildrenParameters, RequestParams } from '../../models/child.model';
+import { DataItem, TruncatedItem } from '../../models/item.model';
 import { PaginationElement } from '../../models/paginationElement.model';
 import { SearchResponse } from '../../models/search.model';
-import { DataItem, TruncatedItem } from '../../models/item.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ChildrenService {
   constructor(private http: HttpClient, private store: Store) {}
 
-  private setParams(parameters: ChildrenParameters, isParent?: boolean): HttpParams {
+  private setParams(parameters: ChildrenParameters): HttpParams {
     let params = new HttpParams();
 
     if (parameters) {
       if (parameters.searchString) {
         params = params.set('SearchString', parameters.searchString);
       }
 
-      if (parameters.isParent !== undefined) {
-        params = params.set('isParent', parameters.isParent.toString());
+      if (typeof parameters.isParent !== 'boolean') {
```

Comment:
The `isParent` can be `boolean` or `null` / `undefined` (possibly).
I don't think it makes sense to pass `isParent=null` to the server, you should pass only this property when the type is `boolean`, and not vice versa.

Reply:
I agree, I got confused in the condition. I'll fix it)